src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -146,6 +147,57 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
                 let tp_ty = substs.type_at(0);
                 self.cx().const_usize(self.cx().size_of(tp_ty).bytes())
             }
+            func @ "va_start" | func @ "va_end" => {
+                let va_list = match tcx.lang_items().va_list() {
+                    Some(did) if result.layout.ty == tcx.type_of(did) => args[0].immediate(),
+                    Some(_) => self.load(args[0].immediate(), tcx.data_layout.pointer_align.abi),
+                    None => bug!("va_list language item must be defined")
+                };
+                let intrinsic = self.cx().get_intrinsic(&format!("llvm.{}", func));
+                self.call(intrinsic, &[va_list], None)
+            }
+            "va_copy" => {
+                let va_list = match tcx.lang_items().va_list() {
+                    Some(did) if result.layout.ty == tcx.type_of(did) => args[0].immediate(),
```

Comment:
I think you should match on `result.layout.ty.sty` for `ty::Adt(def, _)` and compare `def.did` with `did` instead of calling `type_of`.

Reply:
Updated.